.circleci/release.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+: "${GITHUB_TOKEN:?Environment variable GITHUB_TOKEN must be set}"
+: "${CR_REPO_URL:?Environment variable CR_REPO_URL must be set}"
+: "${GIT_USERNAME:?Environment variable GIT_USERNAME must be set}"
+: "${GIT_REPOSITORY_NAME:?Environment variable GIT_REPOSITORY_NAME must be set}"
+
+readonly REPO_ROOT="${REPO_ROOT:-$(git rev-parse --show-toplevel)}"
+export CR_TOKEN="$GITHUB_TOKEN"
+
+main() {
+    pushd "$REPO_ROOT" > /dev/null
+
+    echo "Fetching tags..."
+    git fetch --tags
+
+    local latest_tag
+    latest_tag=$(find_latest_tag)
+
+    local latest_tag_rev
+    latest_tag_rev=$(git rev-parse --verify "$latest_tag")
+    echo "$latest_tag_rev $latest_tag (latest tag)"
+
+    local head_rev
+    head_rev=$(git rev-parse --verify HEAD)
+    echo "$head_rev HEAD"
+
+    if [[ "$latest_tag_rev" == "$head_rev" ]]; then
+        echo "No code changes. Nothing to release."
+        exit
+    fi
+
+    rm -rf .cr-release-packages
+    mkdir -p .cr-release-packages
+
+    rm -rf .cr-index
+    mkdir -p .cr-index
+
+    echo "Identifying changed charts since tag '$latest_tag'..."
+
+    local changed_charts=()
+    local charts_path='*/Chart.yaml' # we consider only charts dirs at level one
+    readarray -t changed_charts <<< "$(git diff --find-renames --name-only "$latest_tag_rev" -- $charts_path | cut -d '/' -f 1 | uniq)"
```

Comment:
I like the idea of releasing only when `Chart.yaml` has changed. :+1: 

What happens if the version has not been changed within `Chart.yaml`?  
I guess `cr` will return an error because of the already existing tag/release. If so, is that error clear enough to allow us to identify the issue quickly?

Reply:
I think this is dope as well. I didn't know about `readarray`. It looks like we'd release on any metadata changes in the charts, but that's ok I suppose.

Reply:
Ideally, we'd release when the version has changed.

Release on any metadata changes would be ok too if `cr` was able to overwrite the GitHub release.

AFAIK `cr` is not able to overwrite an existing tag. Thus I expect an error when `Chart.yaml` was changed without bumping the version number (not 100% sure, maybe we can work it around somehow?)

Reply:
> I like the idea of releasing only when `Chart.yaml` has changed. 
> 
> What happens if the version has not been changed within `Chart.yaml`?
> I guess `cr` will return an error because of the already existing tag/release. If so, is that error clear enough to allow us to identify the issue quickly?

Yes, you guessed it. Without changing also the `version` key after a `Chart.yaml` change you'll get:
```bash
+ cr upload -o xantrix -r test-helm-repo
Error: error creating GitHub release: POST https://api.github.com/repos/xantrix/test-helm-repo/releases: 422 Validation Failed [{Resource:Release Field:tag_name Code:already_exists Message:}]
```
ATM Chart releaser does not support overwriting of an existing tag
https://github.com/helm/chart-releaser/issues/65
but it should be an edge case.

Reply:
Solved by https://github.com/falcosecurity/charts/pull/31/commits/729562a57f4242c346101cb81b0fe7d5380d0c72